test/reduce.jl

```diff
@@ -671,7 +671,7 @@ end
 # issue #38627
 @testset "overflow in mapreduce" begin
     # at len = 16 and len = 1025 there is a change in codepath
-    for len in [0, 1, 15, 16, 1024, 1025, 2048, 2049]
+    for len in [1, 15, 16, 1024, 1025, 2048, 2049]
```

Comment:
Since OffsetArrays more eagerly checks the overflow behavior, `len=0`  is indeed an undefined behavior so I agree to remove it. -- Specially handling length 0 array could also be problematic because user codes might do `push!(parent(offset_vector), x)`.

This `OffsetArray(Int[], typemax(Int))` issue is already caught in OffsetArrays codebase https://github.com/JuliaArrays/OffsetArrays.jl/blob/44f9b50c5428416c8baa76af882c820605c17082/test/runtests.jl#L518

Reply:
The length-0 case is also related to https://github.com/JuliaLang/julia/issues/45389